widgets/filter/src/Filter.tsx

```diff
@@ -210,6 +211,8 @@ export default class Filter extends React.Component {
                                 entityName: 'Deployment',
                                 filter: blueprintFilter,
                                 pageSize: 20,
+                                textFormatter: item => `${item.display_name} (${item.id})`,
```

Comment:
I assume we could easily do

```suggestion
                                textFormatter: item => item.display_name === item.id ? item.id : `${item.display_name} (${item.id})`,
```

to already handle the case when both ID and display name are the same, in which case we want to show only one of them :slightly_smiling_face:

Reply:
Make sense. 

As a side note I'm wondering shouldn't we start with RD-2693 before RD-2324. 
I think that within RD-2693 we could develop a single function which gets `display_name` and `id` values and produces user-friendly text (either `[display_name] ([id])` or just a `[display_name]`). Then we can just reuse it in all the places where we display deployment name and ID.